### PR TITLE
added com.github.rajsolai.response

### PR DESCRIPTION
**Pull request by RajSolai**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
There is no manifest.

**Comment by RajSolai** [comment]
Opps ! Sorry ,Am On it

**Comment by RajSolai** [comment]
Done Adding Manifest file (as yml) !

**Comment by hfiguiere** [comment]
bot, build com.github.rajsolai.response

**Comment by hfiguiere** [comment]
Upstream repository has appdata and desktop file, so why are they in this repository as well? Even more since they are not even referenced in the manifest.

File: com.github.rajsolai.response.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - '--filesystem=home'
+  - '--share=network'
+  - '--device=dri'
+
+modules:
+  - name: response
+    buildsystem: meson
+    sources:
+      - type: archive
+        path: https://github.com/RajSolai/Response/archive/refs/tags/1.0.0.tar.gz
```
Comment (hfiguiere) [inline com.github.rajsolai.response.yml:24]: 1. I think you want `url` here
2. You'll need the `sha256` checksum added.

File: com.github.rajsolai.response.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+base: io.elementary.BaseApp
+base-version: 'juno-20.08'
+sdk: org.gnome.Sdk
+command: com.github.rajsolai.response
+
+finish-args:
+  - '--share=ipc'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+  - '--socket=x11'
```
Comment (hfiguiere) [inline com.github.rajsolai.response.yml:14]: also if you have `wayland`, you shouldn't have `x11` in most cases. (`fallback-x11` is here as a fallback)

Comment (RajSolai) [reply com.github.rajsolai.response.yml:14]: Thanks , On it !

File: com.github.rajsolai.response.yml (hunk 1 above)
Comment (RajSolai) [reply com.github.rajsolai.response.yml:24]: Thanks , Changing now !

**Comment by RajSolai** [comment]
Done , Made changes to manifest

**Comment by hfiguiere** [comment]
Sill the question about the appdata and desktop files:
the upstream repo has them.
the one in this repo aren't referenced, and likely don't need to be here.

**Comment by RajSolai** [comment]
The appdata and desktop files in data dir are for the Elementary OS appcenter. Do I need to remove that one ?

**Comment by RajSolai** [comment]
I removed the flatpak manifest in the mainstream repo !

**Comment by hfiguiere** [comment]
> The appdata and desktop files in data dir are for the Elementary OS appcenter. Do I need to remove that one ?

It is policy to use the appstream and desktop files from upstream.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Use upstream appdata and desktop files. If they need changes, patch them and make sure the patches are submitted upstream.

**Comment by RajSolai** [comment]
Yea , the appcenter uses the Appstream and desktop file to install it's like an boilerplate to have appdata and desktopfile in there for elementary OS appcenter.

**Comment by RajSolai** [comment]
> Use upstream appdata and desktop files. If they need changes, patch them and make sure the patches are submitted upstream.

Sure Thank you , I will do the patches if they require any changes.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Before requesting a review, make sure the requested change are done.

I already talked about the .appdata.xml and .destkop files. Use upstrea and remove the one in the repository here.

**Comment by RajSolai** [comment]
Sorry .. Some misunderstanding I will remove the desktop and appdata.

**Comment by hfiguiere** [comment]
bot, build com.github.rajsolai.response

**Comment by RajSolai** [comment]
Sorry to go out of context , But is there any extra steps for publishing after review or is there any time period for publishing ?

**Comment by nedrichards** [comment]
Once the app passes review a script will create a repo for the manifest and send you an invitation. Most of the other stuff from then on is on the [App Maintenance](https://github.com/flathub/flathub/wiki/App-Maintenance) page linked above.

File: com.github.rajsolai.response.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+app-id: com.github.rajsolai.response
+
+runtime: org.gnome.Platform
+runtime-version: '3.38'
```
Comment (bilelmoussaoui) [inline com.github.rajsolai.response.yml:4]:
```suggestion
runtime-version: '40'
```
maybe?

File: com.github.rajsolai.response.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+base: io.elementary.BaseApp
+base-version: 'juno-20.08'
+sdk: org.gnome.Sdk
+command: com.github.rajsolai.response
+
+finish-args:
+  - '--share=ipc'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+  - '--filesystem=home'
```
Comment (bilelmoussaoui) [inline com.github.rajsolai.response.yml:14]: what does it need home access for?

**Comment by nedrichards** [comment]
As an elementary app I think you might want to patch it to force the elementary stylesheet. Running this on fedora it picks up the default adwaita and gets very low contrast text on the "check" button which I assume is not intentional. Here's an example of doing it: https://github.com/flathub/com.github.muriloventuroso.givemelyrics/blob/ef8251fd657e2472673b4c02001e82b40ee739a4/eos-theme.patch

File: com.github.rajsolai.response.yml (hunk 4 above)
Comment (RajSolai) [reply com.github.rajsolai.response.yml:14]: Opps., Sorry. I'll Remove that.

File: com.github.rajsolai.response.yml (hunk 3 above)
Comment (RajSolai) [reply com.github.rajsolai.response.yml:4]: Am not sure about migrating to gnome 40.. May upcoming releases.

Comment (hfiguiere) [reply com.github.rajsolai.response.yml:4]: why not now? it should just work.

**Review by hfiguiere — COMMENTED** [review]
.

**Comment by nedrichards** [comment]
bot, build com.github.rajsolai.response

File: com.github.rajsolai.response.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+base: io.elementary.BaseApp
+base-version: 'juno-20.08'
+sdk: org.gnome.Sdk
+command: com.github.rajsolai.response
+
+finish-args:
+  - '--share=ipc'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+  - '--share=network'
```
Comment (bilelmoussaoui) [inline com.github.rajsolai.response.yml:14]: you probably want to keep dri permission here, especially that you're using webkit

Comment (RajSolai) [reply com.github.rajsolai.response.yml:14]: Ohh... I tested on my local computer it worked fine btw app does'nt save any cookies with webkit.

File: com.github.rajsolai.response.yml (hunk 3 above)
Comment (barthalion) [reply com.github.rajsolai.response.yml:4]: We're generally not accepting apps using old runtimes unless there's a good reason for it.

File: com.github.rajsolai.response.yml (hunk 5 above)
Comment (barthalion) [inline com.github.rajsolai.response.yml:14]:
```suggestion
  - '--share=network'
  - '--device=dri'
```

**Comment by nedrichards** [comment]
bot, build com.github.rajsolai.response

File: com.github.rajsolai.response.yml (hunk 3 above)
Comment (nedrichards) [reply com.github.rajsolai.response.yml:4]: I tested this using:
> flatpak run --runtime-version=40 com.github.rajsolai.response

and everything seems to work fine. Other than this change everything looks good to merge for me.

**Comment by barthalion** [comment]
bot, build com.github.rajsolai.response

**Comment by barthalion** [comment]
@bilelmoussaoui FYI there's still lots of files lying around and no cleanup script.

**Comment by bilelmoussaoui** [comment]
> @bilelmoussaoui FYI there's still lots of files lying around and no cleanup script.

I'm aware of that, I just didn't have the time to take a look at that. See https://github.com/flathub/io.elementary.BaseApp/issues/5

**Pull request closed without merging**